arch/arm64/core/reset.S

```diff
@@ -89,16 +100,33 @@ out:
 
 #if CONFIG_MP_NUM_CPUS > 1
 L_secondary_stack:
-	get_cpu_id x1
+	get_cpu_id x3
 	adr	x0, arm64_cpu_init
 	mov	x2, #ARM64_CPU_INIT_SIZE
-	madd	x0, x1, x2, x0
-	ldr	x0, [x0]
-	cbz	x0, L_enable_secondary
+	/* Loop and find its table */
+	mov	x1, xzr
+4:
+	madd	x4, x1, x2, x0
+	add	x4, x4, #(ARM64_INIT_TB_MPID_OFFSET)
+	ldr	x4, [x4]
+	cmp	x4, x3
+	dmb	ld
+	b.eq	5f
+	add	x1, x1, #1
+	cmp	x1, #CONFIG_MP_NUM_CPUS
+	b.lt	4b
+	/* can't find related table */
+	b	L_enable_secondary
+5:
+	/* found related table */
+	madd	x4, x1, x2, x0
+	/* Get SP address */
+	add     x4, x4, #(ARM64_INIT_TB_SP_OFFSET)
+	ldr	x4, [x4]
+	cbz	x4, L_enable_secondary
```

Comment:
Not a huge fan of this. I spent a lot of time in the past trying to move al the reset code to C and once again we are back at having a lot of assembly code. Can you try to move this into `smp.c` if there is not a valid technical reason to have it in assembly?

Reply:
these code locates at the place that before set SP, so I don't think we can move it to C.